File: resubmitcrabjobs.pl
#!/usr/bin/perl

use Cwd;

$numArgs = $#ARGV + 1;
if($numArgs < 1)
{
  print(">>>usage: resubmitcrabjobs FOLDER\n");
  exit;
}

$FOLDER = $ARGV[0];

$RESUBMITFLAG = 0;
if($numArgs > 1)
{
  $RESUBMITFLAG = $ARGV[1];
}



print("*************************************************************************************************\n");
print($FOLDER."\n");
print("*************************************************************************************************\n");



# create file with crab report
$REPORTFile = "./report.txt" ;
system("crab -status -c ".$FOLDER." > ".$REPORTFile);



$isExitCodeLine = 0;
$isEventNumberLine = 0;
open(REPORTFILE, $REPORTFile) ;
while(<REPORTFILE>)
{
  chomp;
  s/#.*//;  # no comments
  s/^\s+//; # no leading white
  s/\s+$//; # no trailing white                                                                                                                                      
  
  $line = $_;
  
  if($isEventNumberLine == 1)
  {
    @words = split(" ",$line);
    $eventList = @words[$words-1];
    
    $command = "crab -resubmit ".$eventList." -c ".$FOLDER;
    print($command."\n");
    if($RESUBMITFLAG == 1)
    {
	system($command);
    }
    
    $isEventNumberLine = 0;
  }
  
  if($isExitCodeLine == 1)
  {  
    @words = split(" ",$line);
    $exitCode = @words[$words-1];
    
    if($exitCode != 0)
    {
      print("\nExitCode : ".$exitCode."\n"); 
      $isEventNumberLine = 1;
    }
    
    $isExitCodeLine = 0;
  }
  
  if($isExitCodeLine == 0)
  {
    if( $line eq "crab:  ExitCodes Summary" )
    {
      $isExitCodeLine = 1;
    }
  }
  
}



system("rm ".$REPORTFile);

File: resubmitCrabJobs.py
#!/usr/bin/python

import os
from optparse import OptionParser

parser = OptionParser()
parser.usage = '''
%prog <crabDir> 

crabDir: folder you want to inspect, get output and resubmit failed jobs

'''

parser.add_option("-f", "--forcePending", 
                  dest="force", 
                  help="forceResubmit for pending, stuck, unknown jobs (the ones you cannot getOutput of)",
                  default=False)

parser.add_option("-F", "--forceAll", 
                  dest="forceAll", 
                  help="forceResubmit for all jobs that are not Successfully Done",
                  default=False)

(options,args) = parser.parse_args()

if len(args) != 1:
    parser.print_help()
    sys.exit(1)

crabDir = args[0]

print '#'*(len(crabDir)+10)
print '##  ',crabDir,'  ##'
print '#'*(len(crabDir)+10)

fileName   = os.getcwd()+'/report_'+crabDir+'.txt'
reportFile = open(fileName,'w')

print >> reportFile, 'noknaosjdoai'

os.system('crab -status -c '+crabDir)
os.system('crab -status -c '+crabDir+' >& '+reportFile)



reportFile.close()




